### DISCUSSION TITLE
Bond by distance beyond sequentially connected residues in PDB files?

### DISCUSSION THREAD

**Discussion by gitoliver** [molstar/molstar#823] [Q&A] [answered]
When running MolStar on our website we have branched molecules which results in some of them not having bonds drawn (see the sulfate here):
![Screenshot from 2023-05-19 10-17-59](https://github.com/molstar/molstar/assets/17079588/037fada1-c8b6-4687-99c2-eb2098961248)
There should be a bond between the yellow sulfate atom and the red oxygen.
We can fix this by adding a CONECT card for the atoms in the PDB file, but was wondering if there is a way to get MolStar to check and draw bonds for residues that aren't sequentially adjacent in a pdb file?

**Comment by dsehnal** [comment]
Can you share the PDB file? This might be a bug.

**Reply by gitoliver** [reply]
Sorry got my wires crossed: the conect card fixes the same issue in Chimera, it doesn't fix it in Mol*. I've attached the pdb file but as .txt so I can upload. 
[G13157WZ.pdb.txt](https://github.com/molstar/molstar/files/11518312/G13157WZ.pdb.txt)

**Comment by dsehnal** [comment, marked as answer]
The issue is that Mol* has a special case for ions and doesn't include them in the bond computation by default (the reason for this is performance for large structures). The solution is to turn `dynamicBonds` property on:

![image](https://github.com/molstar/molstar/assets/21145776/7109ccca-c6d5-454b-b132-9c8e455c4372)

This can be done by passing the `dynamicBonds` to the [StructureFromModel](https://github.com/molstar/molstar/blob/master/src/mol-plugin-state/transforms/model.ts#L539-L540) transform.

**Comment by arose** [comment]
> Sorry got my wires crossed: the conect card fixes the same issue in Chimera, it doesn't fix it in Mol*.

That is odd, it should fix it in Mol* too, could you share the conect record?

**Reply by dsehnal** [reply]
I think the issue here is the ION test in the default case and we only check for struct conn records existing between units. 

Perhaps should extend that check?

**Reply by gitoliver** [reply]
Here it is.
[G13157WZ_CONECT.pdb.txt](https://github.com/molstar/molstar/files/11528643/G13157WZ_CONECT.pdb.txt)

**Reply by arose** [reply]
We expect CONECT records in both directions like in archival PDB files and ignore one... So in your example

```
CONECT   11   51
CONECT   51   11
```

or

```
CONECT   11   51
```

will work.

**Reply by gitoliver** [reply]
Ah ok so order matters, I had just 51  11. Thank you.
